var/spack/repos/builtin/packages/util-linux/package.py

```diff
@@ -28,6 +29,10 @@ class UtilLinux(AutotoolsPackage):
     # Make it possible to disable util-linux's libuuid so that you may
     # reliably depend_on(`libuuid`).
     variant('libuuid', default=True, description='Build libuuid')
+    variant('bash', default=False, description='Install bash completion scripts')
+
+    depends_on('bash', when="+bash", type='run')
+    depends_on('libuuid', when="+libuuid")
```

Comment:
This doesn't make sense to me. util-linux ships its own libuuid, why do we want to depend on another implementation?

Reply:
From my understanding of the package, it checks pkg-config first for a version of libuuid that's been installed externally, as that's the preferred method of using libuuid. It only uses the one distributed with the source code as a default if it can't find one already installed on the system.

If I were a developer of util-linux, this makes complete sense. I don't want to force downstream developers to install an explicit dependency, so I put in a fallback in case they haven't installed libuuid yet.  

As a maintainer of spack packages though, why would we ever want to use util-linux's libuuid when clearly the preferred method is to use the libuuid installed on the system?  If libuuid is installed on the system, there's not even any configure flags to use the builtin libuuid over the system libuuid. If this were a desired behavior, certainly there would be a way to turn this behavior on.

Certainly the conflicts in crypt-setup go away if everyone just depends on libuuid, since it depends on libuuid.

Reply:
Sadly, the preferred method doesn't seem to be as clear: https://github.com/spack/spack/issues/10812#issuecomment-590625762
libuuid included in util-linux seems to be the only one that has been updated recently and Linux distributions such as Fedora use util-linux's version.

Reply:
This is also why #18322 makes the util-linux version the default provider.

Reply:
Yep, that's a good catch.  Debian also uses util-linux as the default provider.  Changing the PR.